typegraph/core/src/global_store.rs

```diff
@@ -0,0 +1,149 @@
+// Copyright Metatype OÜ, licensed under the Mozilla Public License Version 2.0.
+// SPDX-License-Identifier: MPL-2.0
+
+use std::{
+    collections::HashMap,
+    sync::{Mutex, MutexGuard},
+};
+
+use once_cell::sync::Lazy;
+
+use crate::{
+    errors::{self, Result},
+    runtimes::{DenoMaterializer, Materializer, MaterializerDenoModule, Runtime},
+    types::{TypeFun, T},
+    wit::core::{Error as TgError, RuntimeId, TypeId},
+    wit::runtimes::{Effect, MaterializerDenoPredefined, MaterializerId},
+};
+
+#[derive(Default)]
+pub struct Store {
+    pub types: Vec<T>,
+    pub type_by_names: HashMap<String, TypeId>,
+
+    pub runtimes: Vec<Runtime>,
+    pub materializers: Vec<Materializer>,
+    default_deno_runtime: Option<RuntimeId>,
```

Comment:
That would be great to add the typegraph spec (see use case here: https://github.com/metatypedev/metatype/pull/338/files#diff-a81d5a26c45ae8197d6468544ed183a9805b40c7d9e53dd1972ef322e8e0101aR233-R241)